CONTRIBUTING.md

```diff
@@ -1,5 +1,96 @@
 # Contributing to qiskit_ibm_transpiler
 
+## Release Notes
+
+When making any end user facing changes in a contribution, we have to make sure
+we document that when we release a new version of qiskit-ibm-transpiler. The
+expectation is that if your code contribution has user facing changes, then you
+will write the release documentation for these changes. This documentation must
+explain what was changed, why it was changed, and how users can either use or
+adapt to the change. The idea behind release documentation is that when a naive
+user with limited internal knowledge of the project is upgrading from the
+previous release to the new one, they should be able to read the release notes,
+understand if they need to update their program which uses qiskit-ibm-transpiler,
+and how they would go about doing that. It ideally should explain why
+they need to make this change too, to provide the necessary context.
+
+To make sure we don't forget a release note or if the details of user facing
+changes over a release cycle we require that all user facing changes include
+documentation at the same time as the code. To accomplish this, we use the
+[Towncrier](https://towncrier.readthedocs.io/en/stable/) tool.
+
+### Adding a new release note
+
+To create a new release note, first find either the issue or PR number associated with
+your change from GitHub because Towncrier links every release note to a GitHub issue
+or PR. If there is no associated issue and you haven't yet opened up the PR so you don't
+yet have a PR number, you can use the value `todo` at first, then go back and rename the
+file once you open up the PR and have its number.
```

Comment:
```suggestion
To create a new release note, first find either the issue or PR number associated with your change from GitHub because Towncrier links every release note to a GitHub issue or PR. If there is no associated issue and you haven't yet opened up the PR so you don't yet have a PR number, you can use the value `todo` at first, then go back and rename the file once you open up the PR and have its number.
```